### DISCUSSION TITLE
is it possible to define a trajectory with low level api?

### DISCUSSION THREAD

**Discussion by lucafei** [microsoft/AirSim#3903] [Support Q&A]
hello, guys, i'm doing a trajectory planning for a aircraft landing, but by no i have only found the airsim api to define the position and velocity. So i want to ask, if it exists another api to plan a custom trajectory or low level api to control the roation of motor?

**Comment by clintlombard** [comment]
Maybe look at running PX4 in SITL with offboard control commands.

**Comment by jonyMarino** [comment]
Hi @lucafei. You can now use the new ExternalPhysicsEngine: #3626

**Comment by lovettchris** [comment]
See moveOnPath in [path.py](https://github.com/microsoft/AirSim/blob/master/PythonClient/multirotor/path.py), it is specifying a path, and velocity, which seems to be exactly what you are asking for?
